cmd/ulxly/ulxly.go

```diff
@@ -126,6 +182,294 @@ var DepositsCmd = &cobra.Command{
+			return err
+		}
+
+		privateKey, err := crypto.HexToECDSA(*ulxlyInputArgs.ClaimPrivateKey)
+		if err != nil {
+			log.Error().Err(err).Msg("Unable to retrieve private key")
+		}
+
+		publicKey := privateKey.Public()
+		publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+		if !ok {
+			log.Error().Msg("Error casting public key to ECDSA")
+		}
+
+		fromAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
+
+		// value := big.NewInt(*ulxlyInputArgs.Value)
+		gasLimit := *ulxlyInputArgs.ClaimGasLimit
+		gasPrice, err := client.SuggestGasPrice(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("Cannot get suggested gas price")
+		}
+
+		toAddress := common.HexToAddress(*ulxlyInputArgs.ClaimAddress)
+
+		chainID, err := client.NetworkID(ctx)
+		if err != nil {
+			log.Error().Err(err).Msg("Cannot get chain ID")
+		}
+
+		signer := types.LatestSignerForChainID(chainID)
+
+		/////////////////////////////////////
+		// This section calls the bridge service RPC URL to get the merkle proofs and exit roots and parses them to the correct formats.
+		bridgeServiceProofEndpoint := fmt.Sprintf("%s/merkle-proof?deposit_cnt=%s&net_id=%s", *ulxlyInputArgs.BridgeServiceRPCURL, *ulxlyInputArgs.ClaimIndex, *ulxlyInputArgs.ClaimNetwork)
+		reqBridgeProof, err := http.Get(bridgeServiceProofEndpoint)
+		if err != nil {
+			log.Error().Err(err)
+		}
+		bodyBridgeProof, err := io.ReadAll(reqBridgeProof.Body) // Response body is []byte
+		if err != nil {
+			log.Error().Err(err)
+		}
+		var bridgeProof BridgeProof
+		err = json.Unmarshal(bodyBridgeProof, &bridgeProof) // Parse []byte to go struct pointer, and shadow err variable
+		if err != nil {
+			log.Error().Err(err).Msg("Can not unmarshal JSON")
+			return err
+		}
+
+		merkleProof := [][32]byte{}
+		rollupMerkleProof := [][32]byte{}
+
+		for _, mp := range bridgeProof.Proof.MerkleProof {
+			byteMP, _ := hexutil.Decode(mp)
+			merkleProof = append(merkleProof, [32]byte(byteMP))
+		}
+		merkleProofArray := [32][32]byte(merkleProof)
+		for _, rmp := range bridgeProof.Proof.RollupMerkleProof {
+			byteRMP, _ := hexutil.Decode(rmp)
+			rollupMerkleProof = append(rollupMerkleProof, [32]byte(byteRMP))
+		}
+		rollupMerkleProofArray := [32][32]byte(rollupMerkleProof)
+
+		mainExitRoot, _ := hexutil.Decode(bridgeProof.Proof.MainExitRoot)
+		rollupExitRoot, _ := hexutil.Decode(bridgeProof.Proof.RollupExitRoot)
+
+		defer reqBridgeProof.Body.Close()
+		/////////////////////////////////////
+
+		tops := &bind.TransactOpts{
+			Signer: func(address common.Address, transaction *types.Transaction) (*types.Transaction, error) {
+				return types.SignTx(transaction, signer, privateKey)
+			},
+			// Value:     value,
+			From:      fromAddress,
+			Context:   ctx,
+			GasLimit:  gasLimit,
+			GasPrice:  gasPrice,
+			GasFeeCap: nil,
+			GasTipCap: nil,
+		}
+
+		/////////////////////////////////////
+		// This section calls the bridge service RPC URL to get the deposits data and parses them to the correct formats.
```

Comment:
This is a bit of a code smell. If your function is so long that you're breaking it up with dividers like this, it's probably worth breaking this down into smaller more understandable functions.

Reply:
+1 you can refactor certain parts of the code, in particular the parts where they share the same logic